genie-proto/src/main/proto/genie.proto

```diff
@@ -264,3 +265,51 @@ message JobKillRegistrationRequest {
 
 message JobKillRegistrationResponse {
 }
+
+//--------------------------------------------------------------------
+// Agent Job File Service
+//
+// This service streams files from an agent running a job to the server
+// so they can be served via API.
+//--------------------------------------------------------------------
+
+service JobFileService {
+    rpc pushManifest (stream JobFileManifestMessage) returns (stream google.protobuf.Empty);
+    rpc streamFile (stream AgentJobFileMessage) returns (stream ServerJobFileMessage);
+}
+
+message JobFileManifestMessage {
+    string job_id = 1;
+    string manifest_json_string = 2;
+}
+
+message AgentJobFileMessage {
+    oneof message {
+        FileStreamRegistration file_stream_registration = 1;
+        FileStreamChunk file_stream_chunk = 2;
+    }
+}
+
+message ServerJobFileMessage {
+    oneof message {
+        FileStreamRequest file_stream_request = 1;
+        FileStreamAck file_stream_ack = 2;
+    }
+}
+
+message FileStreamRegistration {
+    string job_id = 1;
+}
+
+message FileStreamChunk {
+    bytes file_chunk = 2;
```

Comment:
Because of how the prococol works, this is currently not necessary:
1 > Ready
2 < Request(file, offsets)
3 > Chunk
4 < Ack
5 Repeat 3, 4 as needed.
6 > Complete (gRPC primitive to close a stream)

After this, the stream is closed and not reused.

Begin and end message would not send any additional information, but would make the code more complicated. e.g. what if end is received before all data is received?

There's no need for random access on the server. The agent sends chunks in order, and the server is guaranteed to receive them in the same order.
So the server can simply append until all data is received, then stop.